### PR TITLE
Disable Default Target Platform Properties for .NET 5.0+

### PR DESCRIPTION
**Pull request by sfoslund**
Corresponds to microsoft/msbuild#5391

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets [hunk 1]
```diff
@@ -151,6 +151,10 @@ Copyright (c) .NET Foundation. All rights reserved.
       />
   </Target>
 
+  <PropertyGroup  Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 5.0))">
+    <_EnableDefaultWindowsPlatform Condition="'$(_EnableDefaultWindowsPlatform)' == ''" >false</_EnableDefaultWindowsPlatform>
```
Comment (dsplaisted) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets:155]: I think we also wont to set `UseOSWinMdReferences` to false here: https://github.com/microsoft/msbuild/pull/5391#discussion_r434203303

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs [hunk 2]
```diff
@@ -394,7 +394,7 @@ public void It_implicitly_defines_compilation_constants_for_the_target_framework
             definedConstants.Should().BeEquivalentTo(new[] { "DEBUG", "TRACE" }.Concat(expectedDefines).ToArray());
         }
 
-        [Theory]
+        [RequiresMSBuildVersionTheory("16.7.1")]
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs:397]: I think this should just be `16.7`.  VS 16.7 hasn't released yet, and when it does I think the MSBuild version will be 16.7.0.something

Comment (dsplaisted) [reply src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs:397]: Or did microsoft/msbuild#5391 not go into VS 16.7?  In that case I think this value should be `16.8`.

**Review by dsplaisted — APPROVED** [review]
I think with this we could add tests that when the TargetPlatformIdentifier is Windows, the correct compilation constants are defined.

**Comment by sfoslund** [comment]
> I think with this we could add tests that when the TargetPlatformIdentifier is Windows, the correct compilation constants are defined.

@dsplaisted We already have some tests for target platform constants [here](https://github.com/dotnet/sdk/blob/master/src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs#L431), and since we're only defining platform constants for 5.0 and later I don't think we need to change any of the

**Comment by sfoslund** [comment]
Oops, didn't mean to close

**Pull request merged**